### PR TITLE
[RL] Support moe_topk_select using Paddle native operators and Add fused stack-transpose-quant for BlockWiseFP8 MoE weight quantization and swiglu-fp8-quant op for DeepGemmFusedMoE for training alignment

### PR DESCRIPTION
**Pull request by DanielSun11**
## Motivation

> :bulb: If this PR is a Cherry Pick, the PR title needs to follow the format by adding the [Cherry-Pick] label at the very beginning and appending the original PR ID at the end. For example, [Cherry-Pick][CI] Add check trigger and logic(#5191)

> :bulb: 如若此PR是Cherry Pick，PR标题需遵循格式，在最开始加上[Cherry-Pick]标签，以及最后面加上原PR ID，例如[Cherry-Pick][CI] Add check trigger and logic(#5191)

- DeepGemmFusedMoeMethod 的apply_ep_prefill和apply_tp支持使用碎算子组合实现的moe_topk_select，用于和训练对齐。开关为环境变量FD_USE_PHI_MOE_TOPK。
- 权重量化中支持fused_stack_transpose_quant。DeepGemmFusedMoE中支持fuse_weighted_swiglu_fp8_quant
## Checklist

- [x] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [x] Format your code, run `pre-commit` before commit.
- [x] Add unit tests. Please write the reason in this PR if no unit tests.
- [ ] Provide accuracy results.
- [ ] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

**Comment by yuanlehome** [comment]
可以先处理完copilot的review建议～

File: fastdeploy/model_executor/layers/moe/fused_moe_deepgemm_backend.py [hunk 1]
```diff
@@ -164,23 +168,30 @@ def m_grouped_fp8_gemm_nt_contiguous_custom_python_op(
...
-        ffn_in_x, ffn_in_x_scale_tensor = paddle.incubate.nn.functional.fp8_quant_blockwise(
-            ffn_out,
-            using_pow2_scale=not disable_ue8m0_cast,
-            using_ue8m0_scale=not disable_ue8m0_cast,
-        )
-        ffn_in_x_scale_tensor = ffn_in_x_scale_tensor.T[: ffn_in_x.shape[0]]
+        ffn_out = paddle.incubate.nn.functional.swiglu(ffn_out)
+
+        # down_proj
+        if not fastdeploy.envs.FD_USE_PHI_FP8_QUANT:
```
Comment (qingqing01) [inline fastdeploy/model_executor/layers/moe/fused_moe_deepgemm_backend.py:181]: 增加FD_MOE_PROB_IN_ADVANCE开关后，这里的开关是否去掉？

**Comment by yuanlehome** [comment]
FD_USE_PHI_TOPK -> FD_USE_PHI_MOE_TOPK

**Review by qingqing01 — COMMENTED** [review]
FD_USE_PHI_FP8_QUANT
FD_USE_FLEET_FP8_QUANT
FD_MOE_PROB_IN_ADVANCE
1. 这几套能否统一下，现在不确定什么场景应该使用哪套
2. 整体(含permute等)是否用一个环境变量，当前环境变量太多

File: fastdeploy/model_executor/layers/moe/fused_moe_deepgemm_backend.py [hunk 2]
```diff
@@ -196,6 +207,66 @@ def m_grouped_fp8_gemm_nt_contiguous_custom_python_op(
...
+    # group-based topk selection
+    n_group = n_group if n_group > 0 else 1
+    topk_group = topk_group if topk_group > 0 else 1
+    if n_group > 1 and topk_group < n_group:
+        seq_length, n_experts = probs_for_choice.shape
+        group_scores = (
+            probs_for_choice.reshape([seq_length, n_group, -1]).topk(2, axis=-1)[0].sum(axis=-1)
+        )  # [seq_len, n_group]
+        group_idx = paddle.topk(group_scores, k=topk_group, axis=-1, sorted=True)[1]  # [seq_len, topk_group]
+        group_mask = paddle.zeros_like(group_scores).put_along_axis(
```
[the remark this answers is not in the document]
Comment (DanielSun11) [reply fastdeploy/model_executor/layers/moe/fused_moe_deepgemm_backend.py:248]: 对齐训练的api，都使用的put_along_axis，非inplace

File: fastdeploy/model_executor/layers/moe/fused_moe_deepgemm_backend.py [hunk 3]
```diff
@@ -144,6 +147,7 @@ def m_grouped_fp8_gemm_nt_contiguous_custom_python_op(
     layer_added_scale_attrs_1: paddle.Tensor,  # getattr(layer, self.added_scale_attrs[1])
     quant_config_weight_block_size_0: int,  # self.quant_config.weight_block_size[0]
     disable_ue8m0_cast: bool,
+    dst_weights: paddle.Tensor,
```
[the remark this answers is not in the document]
Comment (DanielSun11) [reply fastdeploy/model_executor/layers/moe/fused_moe_deepgemm_backend.py:150]: dst_weights: paddle.Tensor, 不影响infermeta

File: fastdeploy/model_executor/layers/moe/fused_moe_deepgemm_backend.py (hunk 1 above)
Comment (DanielSun11) [reply fastdeploy/model_executor/layers/moe/fused_moe_deepgemm_backend.py:181]: 还是需要留着的。FD_MOE_PROB_IN_ADVANCE 仅仅是代表将乘probs的过程提前默认不开启，FD_USE_PHI_FP8_QUANT这个开关在还是有效的

File: fastdeploy/envs.py [hunk 4]
```diff
@@ -202,6 +202,8 @@ def _validate_split_kv_size(value: int) -> int:
     "FD_USE_PHI_FP8_QUANT": lambda: bool(int(os.getenv("FD_USE_PHI_FP8_QUANT", "1"))),
     # Whether to use phi MOE permute,if 1,use paddle default.
     "FD_USE_PHI_MOE_PERMUTE": lambda: bool(int(os.getenv("FD_USE_PHI_MOE_PERMUTE", "1"))),
+    # Whether to use phi TopK, when need select moe by noaux_tc method if 1,use paddle default.
```
[the remark this answers is not in the document]
Comment (DanielSun11) [reply fastdeploy/envs.py]: done

File: fastdeploy/model_executor/layers/moe/fused_moe_deepgemm_backend.py [hunk 5]
```diff
@@ -196,6 +196,66 @@ def m_grouped_fp8_gemm_nt_contiguous_custom_python_op(
...
+    # group-based topk selection
+    n_group = n_group if n_group > 0 else 1
+    topk_group = topk_group if topk_group > 0 else 1
+    if n_group > 1 and topk_group < n_group:
+        seq_length, n_experts = probs_for_choice.shape
+        group_scores = (
+            probs_for_choice.reshape([seq_length, n_group, -1]).topk(2, axis=-1)[0].sum(axis=-1)
+        )  # [seq_len, n_group]
+        group_idx = paddle.topk(group_scores, k=topk_group, axis=-1, sorted=True)[1]  # [seq_len, topk_group]
+        group_mask = paddle.zeros_like(group_scores).put_along_axis(
```
[the remark this answers is not in the document]
Comment (DanielSun11) [reply fastdeploy/model_executor/layers/moe/fused_moe_deepgemm_backend.py:248]: 保持和训练调用的api一致，暂时不用inplace代替

**Comment by DanielSun11** [comment]
> FD_USE_PHI_TOPK -> FD_USE_PHI_MOE_TOPK

已经更改

**Comment by DanielSun11** [comment]
> FD_USE_PHI_FP8_QUANT FD_USE_FLEET_FP8_QUANT FD_MOE_PROB_IN_ADVANCE
> 
> 1. 这几套能否统一下，现在不确定什么场景应该使用哪套
> 2. 整体(含permute等)是否用一个环境变量，当前环境变量太多

FD_USE_PHI_FP8_QUANT和FD_USE_FLEET_FP8_QUANT 已经合并

**Review by yuanlehome — APPROVED** [review]

**Review by qingqing01 — APPROVED** [review]

**Review by Deleter-D — APPROVED** [review]
